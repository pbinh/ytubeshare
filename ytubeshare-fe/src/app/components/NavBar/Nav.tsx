import * as React from 'react';
import { UserOutlined } from '@ant-design/icons';
import { Button, Col, Input, Row, Space } from 'antd';


export const Nav = ({onLoginRegisterBtnClicked}) => {
  const style = { width: '100%'}
  const [username, setUsername] = React.useState('')
  const [password, setPassword] = React.useState('')

  return (
    <Row gutter={9}>
      <Col span={7}>
        <Input value={username} onChange={(e) => setUsername(e.target.value)} placeholder="Username" prefix={<UserOutlined />} />
      </Col>
      <Col span={7}>
        <Input.Password value={password} onChange={(e) => setPassword(e.target.value)} placeholder="Password" />
      </Col>
      <Col span={10}>
        <Button onClick={() => onLoginRegisterBtnClicked(username, password)} style={style} type="primary">Login / Register</Button>
      </Col>
    </Row>
  );
}

