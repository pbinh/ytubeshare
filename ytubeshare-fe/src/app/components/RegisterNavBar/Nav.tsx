import * as React from 'react';
import { UserOutlined } from '@ant-design/icons';
import { Button, Col, Input, Row, Space } from 'antd';


export const Nav = () => {
  const style = { width: '100%'}
  const [username, setUsername] = React.useState('')
  const [password, setPassword] = React.useState('')

  return (
    <Row gutter={9}>
    </Row>
  );
}

