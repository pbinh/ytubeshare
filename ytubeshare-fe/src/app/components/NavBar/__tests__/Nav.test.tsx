import * as React from 'react';
import { render } from '@testing-library/react';
import { Nav } from '../Nav';
import { createRenderer } from 'react-test-renderer/shallow';

const shallowRenderer = createRenderer();

describe('<Nav />', () => {
  it('should match the snapshot', () => {
    shallowRenderer.render(
      <Nav 
        onLoginRegisterBtnClicked={() => {}} 
        />
    );
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
