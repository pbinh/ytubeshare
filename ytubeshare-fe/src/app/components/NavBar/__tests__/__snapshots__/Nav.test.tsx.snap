// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should match the snapshot 1`] = `
<Row
  gutter={9}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Col
    lg={18}
    md={18}
    sm={16}
    xs={0}
  >
    <div
      className="flex-center-end"
    >
      <Input
        onChange={[Function]}
        placeholder="Username"
        prefix={<ForwardRef(UserOutlined) />}
        value=""
      />
      <Password
        onChange={[Function]}
        placeholder="Password"
        style={
          Object {
            "marginLeft": 10,
          }
        }
        value=""
      />
    </div>
  </Col>
  <Col
    className="flex-center-end"
    lg={6}
    md={6}
    sm={8}
    xs={24}
  >
    <Button
      onClick={[Function]}
      type="primary"
    >
      Login / Register
    </Button>
  </Col>
</Row>
`;
